sql/mysqld.cc

```diff
@@ -2517,8 +2538,13 @@ static void network_init(void)
   struct sockaddr_un	UNIXaddr;
   int	arg;
 #endif
+  int systemd_n = 0;
   DBUG_ENTER("network_init");
 
+#ifdef HAVE_SYSTEMD
+  systemd_n = sd_listen_fds(0);
+  DBUG_PRINT("general",("Systemd listen_fds is %d",systemd_n));
+#endif
```

Comment:
let's try to reduce number if #ifdef's.
in this case, sd_listen_fds() can be defined to 0 without HAVE_SYSTEMD, sd_notify() — to a nop